src/JsonApiSerializer/JsonConverters/ResourceObjectConverter.cs

```diff
@@ -149,146 +167,238 @@ public override void WriteJson(JsonWriter writer, object value, JsonSerializer s
             }
         }
 
-
-        protected void WriteFullObjectJson(JsonWriter writer, object value, JsonSerializer serializer)
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private void WriteFullObjectJson(JsonWriter writer, object value, JsonSerializer serializer)
         {
             var valueType = value.GetType();
             var contractResolver = serializer.ContractResolver;
 
             if (!(contractResolver.ResolveContract(valueType) is JsonObjectContract contract
                  && contract.Converter is ResourceObjectConverter))
-                throw new JsonApiFormatException(writer.Path, 
+                throw new JsonApiFormatException(writer.Path,
                     $"Expected to find to find resource object, but found '{value}'",
-                    "Resource indentifier objects MUST contain 'id' members");
+                    "Resource identifier objects MUST contain 'id' members");
 
-            //prepare to start capturing attributes and relationships
-            var attributes = new List<KeyValuePair<JsonProperty, object>>();
-            var relationships = new List<KeyValuePair<JsonProperty, object>>();
+            // prepare to start capturing attributes and relationships
 
-            //will capture id and type as we go through
-            object id = null;
-            object type = null;
+            // default list size to the number of expected properties 

```

Comment:
comment doesnt match the code. we do not set it to the expected number of properties, we set it to null